### DISCUSSION TITLE
Is it possible to create a file in a directory? (limited file system access)

### DISCUSSION THREAD

**Discussion by romanusname** [swc-project/swc#7421] [Q&A] [answered]
### wasm files have limited access to the file system.

**I can create a file.**
`fs::write("test.txt", "context").expect("error");`

**But I can't create a file in the directory.**
`fs::write("myDirectory/test.txt", "context").expect("error");`

**Also I can't create a directory.**
`fs::create_dir("newDirectory").expect("error");`

I get an error - `Unable to write file: Os { code: 76, kind: Uncategorized, message: "Capabilities insufficient" }`

Is it possible to create a file in a directory?
I need to create many files. I want to group them in a separate directory

**Comment by kwonoj** [comment, marked as answer]
You asked same q already, https://github.com/swc-project/swc/discussions/4997#discussioncomment-5822014 

This is all you got. We expose current cwd as virtual mounted to allow filesystem access, anything it cannot provide means pluing cannot have those capabilities.
